packages/next/server/config.ts

```diff
@@ -234,6 +234,40 @@ function assignDefaults(userConfig: { [key: string]: any }) {
         )
       }
     }
+
+    const remotePatterns = result.experimental?.images?.remotePatterns
+    if (remotePatterns) {
+      if (!Array.isArray(remotePatterns)) {
+        throw new Error(
+          `Specified images.remotePatterns should be an Array received ${typeof remotePatterns}.\nSee more info here: https://nextjs.org/docs/messages/invalid-images-config`
+        )
+      }
+
+      if (remotePatterns.length > 50) {
+        throw new Error(
+          `Specified images.remotePatterns exceeds length of 50, received length (${remotePatterns.length}), please reduce the length of the array to continue.\nSee more info here: https://nextjs.org/docs/messages/invalid-images-config`
+        )
+      }
+
+      const validProps = new Set(['protocol', 'hostname', 'pathname', 'port'])
+      const invalidIndex = remotePatterns.findIndex(
```

Comment:
Should we show all invalid items instead of just the first so that they can all be fixed at the same time? Related check for i18n can be seen here https://github.com/vercel/next.js/blob/3e78f0cb5d9644fc5af6147d96b210c6c0d308bd/packages/next/server/config.ts#L554

Reply:
Perhaps. I just thought it would be really verbose if you had more than one invalid object in the array

Reply:
Done!